sklearnex/covariance/incremental_covariance.py

```diff
@@ -0,0 +1,128 @@
+# ===============================================================================
+# Copyright 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+import numpy as np
+from sklearn.utils import check_array, gen_batches
+
+from onedal.covariance import (
+    IncrementalEmpiricalCovariance as onedal_IncrementalEmpiricalCovariance,
+)
+
+
+class IncrementalEmpiricalCovariance:
+    """Incremental estimator for covariance.
+    Allows to compute empirical covariance estimated by maximum
+    likelihood method if data are splitted into batches.
+
+    Parameters
+    ----------
+    batch_size : int, default=None
+        The number of samples to use for each batch. Only used when calling
+        ``fit``. If ``batch_size`` is ``None``, then ``batch_size``
+        is inferred from the data and set to ``5 * n_features``, to provide a
+        balance between approximation accuracy and memory consumption.
+
+    Attributes
+    ----------
+    location_ : ndarray of shape (n_features,)
+        Estimated location, i.e. the estimated mean.
+
+    covariance_ : ndarray of shape (n_features, n_features)
+        Estimated covariance matrix
+
+    _need_to_finalize : bool
+        If True then finalize compute should be called to
+        obtain covariance_ or location_ from partial
+        compute data
```

Comment:
Do we need this for api?

Reply:
i don't understand the question, clarify please. the usage of this attribute is described in the doc string

Reply:
This attribute is private, non-public.

Due to : [PEP8 documenting](https://peps.python.org/pep-0008/#documentation-strings)

> Write docstrings for all public modules, functions, classes, and methods. Docstrings are not necessary for non-public methods, but you should have a comment that describes what the method does. This comment should appear after the "def" line.

Also applicable for attributes.
So it is better to leave just comment, not a docstring description for a such non-public attribute.